### PR TITLE
Add homepage, post page and archives page

### PR DESCRIPTION
**Pull request by solazio**
## Done

- Add home page
- Add post page
- Add archives page

## QA

- Go to [http://0.0.0.0:8035/](http://0.0.0.0:8035/), [/archives](http://0.0.0.0:8035/archives) & [/ubuntu-updates-to-mitigate-new-microarchitectural-data-sampling-mds-vulnerabilities](http://0.0.0.0:8035/ubuntu-updates-to-mitigate-new-microarchitectural-data-sampling-mds-vulnerabilities)
- Compare to [design home page](https://user-images.githubusercontent.com/36884067/60580917-cef2c400-9d7d-11e9-8a84-6a6e79bd30dd.png), [design archives](https://github.com/canonical-web-and-design/web-squad/issues/1272) & [design post page](https://github.com/canonical-web-and-design/web-squad/issues/1341)

Fixes https://github.com/canonical-web-and-design/web-squad/issues/1276 
Fixes https://github.com/canonical-web-and-design/web-squad/issues/1341
Fixes https://github.com/canonical-web-and-design/web-squad/issues/1275

### CODE REVIEW

**Comment by pmahnke** [comment]
image

- [ ] remove square brackets from elipsis 
- [ ] end shortened blurb on a word

image

- [ ] 'Share on' should be centre aligned with the icons
- [ ] icons aren't loading in non-hovered state correctly

image

- [ ] legal link should go to https://ubuntu.com/legal
- [ ] data privacy link should got to https://ubuntu.com/legal/data-privacy
- [ ] bug link should go to https://github.com/canonical-web-and-design/kubeflow-news.com/issues/new?body=%0a%0a%0a---%0a*Reported%20from:%20 ADD DYNAMIC URL HERE*

image

- [ ] is there any alignment issue here?

image

- [ ] I don't thnk we need an author

OTHER

- [ ] Latest doesn't work
- [ ] What's the success message for signing up?

File: package.json [hunk 1]
```diff
@@ -1,4 +1,6 @@
 {
```
Comment (tbille) [inline package.json]: can you change this line to fix th css build:

`"build": "node-sass --include-path node_modules static/sass --source-map true --output-style compressed --output static/css && postcss --map false --use autoprefixer --replace 'static/css/**/*.css'",`

File: templates/blog/archives.html [hunk 2]
```diff
@@ -3,15 +3,57 @@
 {% block title %}Archives{% endblock %}
 
 {% block content %}
-  <h1>Archives</h1>
+  <header class="p-strip--dark p-strip--newsletter" style="text-align: left">
+    <div class="row">
+      <div class="col-10">
+        <h1>Archives</h1>
+        <p>
+          {{ total_posts }} post{{ total_posts|pluralize }}
```
Comment (jpmartinspt) [inline templates/blog/archives.html]: Not sure it will happen, but if total posts is 0 it will read "0 post"

**Comment by solazio** [comment]
@sowasred2012 I think you got the navigation button colours wrong ... at least according to [this image](https://user-images.githubusercontent.com/36884067/60581481-d070bc00-9d7e-11e9-9130-a97fa281021a.png). The selected page is supposed to be black and the other ones grey?

**Comment by solazio** [comment]
@pmahnke @anasereijo @spencerbygraves this is ready to be checked.

**Comment by anasereijo** [comment]
## Post page:

- [ ] align social share to the right (edge of the page)

- [ ] `margin-bottom:0;` on social media share (so it matches the top margin)

Mobile:

- [ ] align social icons to the left edge of the page

Screenshot 2019-07-08 at 16 24 56

- [ ] remove  `u-no-margin--bottom` from the title on mobile. It looks too close so it will be hard to read on small screens

- [ ] add `hr` line separator between post and form

Screenshot 2019-07-08 at 16 35 17


## Archives:

- [ ] number of posts should be `h4`

- [ ] "see all" button is too close to the link above. Can you give it a bit more padding? 40px would be ideal but happy to check with you what works best with vanilla.

File: templates/_footer.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    </div>
+  </div>
+</footer>
+
+<div class="col-7">
+
+    
+    
+    </nav>
+  </div>
```
Comment (tbille) [inline templates/_footer.html]: I think that this block is a mistake :smile:

**Comment by anasereijo** [comment]
Nice! 🥇

File: templates/blog/_post.html [hunk 4]
```diff
@@ -1,6 +1,22 @@
...
+    <div class="row">
+      <div class="col-7 p-post">
+        {{ article.content.rendered|safe }}
+        
+        {% for related_article in related_articles %}
+          <hr class="p-separator" />
+          {% include 'blog/_media-object.html' %}
+          {% if loop.last %}
+            <hr class="p-separator__mobile" />
+          {% endif %}
```
Comment (tbille) [inline templates/blog/_post.html]: I think this block can be removed and added after the `for` block. Like this we avoid a if for every iteration in the loop

File: static/sass/_pattern_buttons.scss [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    padding-right: 1.8rem;
+
+    &:hover,
+    &:active:hover {
+      background-color: #1257f1;
+      border-color: #1257f1;
+    }
+
+    &:visited {
+      color: #fff;
```
Comment (anthonydillon) [inline static/sass/_pattern_buttons.scss]: Can you use `$color-x-light`

File: static/sass/_pattern_footer.scss [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  .p-footer {
+    background-color: $color-brand;
+
+    &__logo {
+      margin-right: 0.5rem;
+    }
+
+    .p-inline-list--middot {
+      .p-inline-list__item::after {
+        color: #fff;
```
Comment (anthonydillon) [inline static/sass/_pattern_footer.scss]: `$color-x-light`

File: templates/blog/_newsletter-signup.html [hunk 7]
```diff
@@ -1,29 +1,40 @@
-<div>
-  <header>
-    <h2>Newsletter Signup</h2>
-  </header>
+<section class="p-strip--dark is-deep p-strip--newsletter">
```
Comment (anthonydillon) [inline templates/blog/_newsletter-signup.html]: You could use `p-strip--image` from Vanilla

File: static/sass/_settings_colors.scss [hunk 8]
```diff
@@ -0,0 +1,5 @@
+$color-brand:             #213d7a;;
```
Comment (anthonydillon) [inline static/sass/_settings_colors.scss]: Remove the double ;

File: templates/blog/_newsletter-signup-form.html [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+      <input class="u-no-margin--bottom" id="Email" placeholder="Work email" name="Email" maxlength="255" type="email" required />
+    </div>
+    <p>
+      <small>In submitting this form, I confirm that I have read and agree to <a href="/legal/data-privacy/newsletter">Canonical’s Privacy Notice</a> and <a href="/legal/data-privacy">Privacy Policy</a>.</small>
+    </p>
+    <p class="u-no-margin--bottom">
+      <input type="hidden" name="Consent_to_Processing__c" value="Yes" />
+      <button class="p-button" type="submit">Subscribe now</button>
+    </p>
+    <input type="hidden" name="formid" value="3415" />
```
Comment (anthonydillon) [inline templates/blog/_newsletter-signup-form.html]: Is this the correct formid

Comment (sowasred2012) [reply templates/blog/_newsletter-signup-form.html]: Yep, @caldav set a new one up for us and passed the ID on.

**Pull request merged**